datatable/java/datatable/src/test/java/io/cucumber/datatable/DataTableTypeRegistryTest.java

```diff
@@ -120,4 +120,24 @@ public void returns_entry_data_table_type() {
 
         assertSame(entry, lookupTableTypeByType);
     }
+
+    @Test
+    public void parse_decimal_with_english_locale() {
+        expectDecimalValueForDecimalStringInLocale(Locale.ENGLISH, 12.45, "12.45");
+    }
+
+    @Test
+    public void parse_decimal_with_polish_locale() {
+        expectDecimalValueForDecimalStringInLocale(Locale.forLanguageTag("pl"), 12.45, "12,45");
+
+    }
+
+    private void expectDecimalValueForDecimalStringInLocale(Locale locale, double expectedDecimalValue, String decimalString) {
+        DataTableTypeRegistry typeRegistry = new DataTableTypeRegistry(locale);
+        //noinspection unchecked
+        List<List<BigDecimal>> transform = (List<List<BigDecimal>>) typeRegistry.lookupTableTypeByType(LIST_OF_LIST_OF_DECIMAL).transform(Collections.singletonList(Collections.singletonList(decimalString)));
+
+        assertEquals(1,transform.size());
+        assertEquals(0,transform.get(0).get(0).compareTo(BigDecimal.valueOf(expectedDecimalValue)));
```

Comment:
This again causes a loss of precision. Rather then creating the expected `BigDecimal` from a `double` a test should always create `BigDecimals` directly from string.

Note that this error is extra sneaky because it is hiding in all the setup code.